sematic/ui/packages/common/src/pages/RunSearch/SearchFilters.tsx

```diff
@@ -80,8 +82,20 @@ const SearchFilters = (props: SearchFiltersProps) => {
         onFiltersChanged({...allFilters.current});
     }, [onFiltersChanged]);
 
+    const [submitedDefaults, setSubmitedDefaults] = useState(false);
+    useEffect(() => {
+        if(submitedDefaults) {
+            return;
+        }
+        if(!!defaultSearchString) {
+            allFilters.current[FilterType.SEARCH] = [defaultSearchString];
+        }
+        applyFilters();
+        setSubmitedDefaults(true);
+    }, [submitedDefaults, setSubmitedDefaults, applyFilters, allFilters, defaultSearchString]);
```

Comment:
We can always empty the dependency array if it is a one-off operation. 

Are you using `submitedDefaults` to avoid frequent updates?

Reply:
after the refactor to address the double-search, this is irrelevant